controllers/gateway_controller.go

```diff
@@ -254,9 +250,6 @@ func (r *GatewayReconciler) reconcileGatewayResources(ctx context.Context, gw *g
 	if err != nil {
 		return err
 	}
-	if snInfo == nil {
-		return fmt.Errorf("Service network %s for account %s not found", gw.Name, config.AccountID)
-	}
```

Comment:
With the change on ```FindX``` logic to raise a "not found" error, these additional nil checks on the return value are no longer required.